File: event-schemas/src/schemas/eventValues/transaction.ts
/* eslint-disable @typescript-eslint/no-unused-vars */
import {
  JsonSchema, JsonSchemaType
} from '../../json-schema';

import { StringRef, NumberRef } from "../types";

export const Transaction: JsonSchema = {
  title: 'Transaction',
  type: JsonSchemaType.OBJECT,
  description: 'The Transaction Payload. All the values for Transaction interactions',
  properties: {
    id: StringRef,
    receipt: StringRef,
    receiptSignature: StringRef,
    transactorId: StringRef,
    transactorName: StringRef,
    transactorType: StringRef,
    receiverId: StringRef,
    receiverName: StringRef,
    receiverType: StringRef,
    network: StringRef,
    checkoutId: StringRef,
    orderId: StringRef,
    storeId: StringRef,
    storeSourceId: StringRef,
    renewalCount: { type: JsonSchemaType.NUMBER, default: 0 },
    feesTotal: NumberRef,
    priceTotal: NumberRef,
    discountTotal: NumberRef,
    quantity: NumberRef,
    amount: NumberRef,
    currency: StringRef,
    paymentMethod: StringRef,
    paymentProvider: StringRef,

    virtualAmount: NumberRef,
    virtualCurrency: StringRef,
    virtualCurrencyType: StringRef,

    blob: StringRef,
  }
}

File: event-schemas/data/golang/Transaction.go

package entities

// Transaction represents a Transaction model.
type Transaction struct {
  Id string
  Receipt string
  ReceiptSignature string
  TransactorId string
  TransactorName string
  TransactorType string
  ReceiverId string
  ReceiverName string
  ReceiverType string
  Network string
  CheckoutId string
  OrderId string
  StoreId string
  StoreSourceId string
  RenewalCount float64
  FeesTotal float64
  PriceTotal float64
  DiscountTotal float64
  Quantity float64
  Amount float64
  Currency string
  PaymentMethod string
  PaymentProvider string
  VirtualAmount float64
  VirtualCurrency string
  VirtualCurrencyType string
  Blob string
  AdditionalProperties map[string][]interface{}
}